includes/wp-cli.php

```diff
@@ -3,6 +3,48 @@
 class WPCOM_Legacy_Redirector_CLI extends WP_CLI_Command {
 
 	/**
+	 * Find domains redirected to, useful to populate the allowed_redirect_hosts filter.
+	 *
+	 * @subcommand find-domains
+	 */
+	function find_domains( $args, $assoc_args ) {
+		$posts_per_page = 500;
+		$paged = 1;
+
+		$domains = array();
+		$total_redirects = wp_count_posts( WPCOM_Legacy_Redirector::POST_TYPE );
+
+		$progress = \WP_CLI\Utils\make_progress_bar( 'Finding domains', array_sum( (array) $total_redirects ) );
+		do {
+			$posts = get_posts( array(
+				'post_type' => WPCOM_Legacy_Redirector::POST_TYPE,
+				'posts_per_page' => $posts_per_page,
+				'paged'          => $paged,
+			) );
+
+			foreach ( $posts as $post ) {
+				$progress->tick();
+				if ( '' !== $post->post_excerpt ) {
+					if ( wp_parse_url( $post->post_excerpt, PHP_URL_HOST ) ) {
+						$domains[] = wp_parse_url( $post->post_excerpt, PHP_URL_HOST );
```

Comment:
Given that `wp_parse_url()` might return `false` and to avoid using it twice, how about assigning the result to a variable? Then `$domains` need only be added to when there is definitely something to add.